Unable to build in Android: unreported exception JSONException

**Description:**
Unable to build android app:

/Users/fabio/Documents/Progetti/iOS/country-italia-app/node_modules/react-native-onesignal/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:489: error: unreported exception JSONException; must be caught or declared to be thrown
               callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
                                                                                  ^
/Users/fabio/Documents/Progetti/iOS/country-italia-app/node_modules/react-native-onesignal/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:502: error: unreported exception JSONException; must be caught or declared to be thrown
               callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
                                                                                  ^
/Users/fabio/Documents/Progetti/iOS/country-italia-app/node_modules/react-native-onesignal/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java:515: error: unreported exception JSONException; must be caught or declared to be thrown

**Environment**
React Native: 0.62.2

buildToolsVersion = "28.0.3"
minSdkVersion = 21
compileSdkVersion = 29
targetSdkVersion = 29

react-native-onesignal: 3.8.0

**Steps to Reproduce Issue:**
Try to build

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -490,7 +490,6 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
                   callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
                } catch (JSONException e) {
                   Log.e("OneSignal", "sendOutcome with name: " + name + ", failed with message: " + e.getMessage());
-                  e.printStackTrace();
                }
             }
          }
@@ -509,7 +508,6 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
                   callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
                } catch (JSONException e) {
                   Log.e("OneSignal", "sendUniqueOutcome with name: " + name + ", failed with message: " + e.getMessage());
-                  e.printStackTrace();
                }
             }
          }
@@ -528,7 +526,6 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
                   callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
                } catch (JSONException e) {
                   Log.e("OneSignal", "sendOutcomeWithValue with name: " + name + " and value: " + value + ", failed with message: " + e.getMessage());
-                  e.printStackTrace();
                }
             }
          }

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -479,40 +479,58 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
     */
 
    @ReactMethod
-   public void sendOutcome(String name, final Callback callback) {
+   public void sendOutcome(final String name, final Callback callback) {
       OneSignal.sendOutcome(name, new OutcomeCallback() {
          @Override
          public void onSuccess(OutcomeEvent outcomeEvent) {
             if (outcomeEvent == null)
                callback.invoke(new WritableNativeMap());
-            else
-               callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+            else {
+               try {
+                  callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+               } catch (JSONException e) {
+                  Log.e("OneSignal", "sendOutcome with name: " + name + ", failed with message: " + e.getMessage());
+                  e.printStackTrace();
+               }
+            }
          }
       });
    }
 
    @ReactMethod
-   public void sendUniqueOutcome(String name, final Callback callback) {
+   public void sendUniqueOutcome(final String name, final Callback callback) {
       OneSignal.sendUniqueOutcome(name, new OutcomeCallback() {
          @Override
          public void onSuccess(OutcomeEvent outcomeEvent) {
             if (outcomeEvent == null)
                callback.invoke(new WritableNativeMap());
-            else
-               callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+            else {
+               try {
+                  callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+               } catch (JSONException e) {
+                  Log.e("OneSignal", "sendUniqueOutcome with name: " + name + ", failed with message: " + e.getMessage());
+                  e.printStackTrace();
+               }
+            }
          }
       });
    }
 
    @ReactMethod
-   public void sendOutcomeWithValue(String name, float value, final Callback callback) {
+   public void sendOutcomeWithValue(final String name, final float value, final Callback callback) {
       OneSignal.sendOutcomeWithValue(name, value, new OutcomeCallback() {
          @Override
          public void onSuccess(OutcomeEvent outcomeEvent) {
             if (outcomeEvent == null)
                callback.invoke(new WritableNativeMap());
-            else
-               callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+            else {
+               try {
+                  callback.invoke(RNUtils.jsonToWritableMap(outcomeEvent.toJSONObject()));
+               } catch (JSONException e) {
+                  Log.e("OneSignal", "sendOutcomeWithValue with name: " + name + " and value: " + value + ", failed with message: " + e.getMessage());
+                  e.printStackTrace();
+               }
+            }
          }
       });
    }

